### PR TITLE
MM-41752: Batch update mention increment

### PR DESCRIPTION
**Pull request by agnivade**
Previously, we were incrementing mentions one-by-one
all concurrently in an unbounded fashion.

This would cause a big spike in memory usage if there
were an `@all` mention in a large channel.

We fix this by changing the SQL query to take all userIDs
at once.

https://mattermost.atlassian.net/browse/MM-41752

```release-note
NONE
```

### CODE REVIEW

File: store/storetest/user_store.go
Comment (larkox) [inline store/storetest/user_store.go:2339]: I don't see tests with more than one id (they should work fine, but still, would be nice to have).

Also, just as a sanity check, it would be nice to have a test with the same id repeated. Looking at the code it should be solved fine, but seems a nice to have too.

**Review by larkox — APPROVED** [review]
LGTM, just one small proposal for tests.

File: store/storetest/user_store.go
Comment (agnivade) [reply store/storetest/user_store.go:2339]: Yep, good point. I am just waiting for Jesse to return and merge the other PR on which it's blocked.

Comment (lieut-data) [reply store/storetest/user_store.go:2339]: I've resolved the merge conflicts and should auto-merge https://github.com/mattermost/mattermost-server/pull/19523 shortly.

Comment (agnivade) [reply store/storetest/user_store.go]: Tests added!

**Comment by agnivade** [comment]
@ashishbhate @koox00 - ready for review.

File: store/storetest/user_store.go
Comment (koox00) [inline store/storetest/user_store.go:2338]: Wouldn't the actual scenario be to increment both at the same time?

```suggestion
	nErr = ss.Channel().IncrementMentionCount(c1.Id, []string{u2.Id, u3.Id}, false)
```

File: store/storetest/user_store.go
Comment (larkox) [reply store/storetest/user_store.go]: Maybe I am missing something, but I don't see how the new tests cover the two cases I noted (more than one id, and repeated id).

File: store/storetest/user_store.go
Comment (agnivade) [reply store/storetest/user_store.go:2338]: Oh, brain.exe stopped working for a while there.

File: store/storetest/user_store.go
Comment (agnivade) [reply store/storetest/user_store.go]: PTAL :)

**Review by koox00 — APPROVED** [review]

**Review by lieut-data — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @agnivade 
Tested, looks good to merge.
- Verified mentions are working as expected.
- [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/157780-5b19d29-onprem-ent-server-pr-19596-mattermost/mm-ee-test5b19d29/mochawesome.html) looks good, no PR related failures.

**Comment by agnivade** [comment]
@ashishbhate - Going to merge this since I have 4 dev approvals.

**Review by ashishbhate — COMMENTED** [review]
LGTM! 

Sorry about the delay in getting to this, @agnivade

**Comment by amyblais** [comment]
@agnivade This can be manually cherry-picked to ``release-6.3``.

**Comment by agnivade** [comment]
Ahh sorry @amyblais - this PR is built on top of some refactoring work that already went in for CRT. To make this work with 6.3 would mean to apply all of those previous work as well. Let me know if you think it's important enough to do that.

**Comment by amyblais** [comment]
> Ahh sorry @amyblais - this PR is built on top of some refactoring work that already went in for CRT. To make this work with 6.3 would mean to apply all of those previous work as well. Let me know if you think it's important enough to do that.

Sounds good. I think it's not needed to be done.

**Pull request merged**
